lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -551,10 +556,129 @@ def build_module(self, trans, payload={}):
             'config_form'       : module.get_config_form(),
             'post_job_actions'  : module.get_post_job_actions(inputs)
         }
+        
+    @expose_api
+    def get_tool_predictions(self, trans, topk=10, max_seq_len=25, payload={}):
+        """
+        POST /api/workflows/get_tool_predictions
+        Fetch predicted tools for a workflow
+        """
+        trained_model_path = trans.app.config.model_path
+        tool_sequence = payload.get('tool_sequence', "")
+        if 'tool_sequence' not in payload or trained_model_path is None:
+            return
+        else:
+            r_file = requests.get(trained_model_path)
+            model_save_path = os.path.join(os.getcwd(), "database/trained_model.hdf5")
+            if not os.path.exists(model_save_path):
+                with open(model_save_path, 'wb') as fl:
+                    fl.write(r_file.content)
+            m_file = h5py.File(model_save_path, 'r')
+            toolbx = trans.app.toolbox.to_dict(trans, in_panel=True)
```

Comment:
you might be able to work with this without the `to_dict` call? I imagine that adds some overhead to calling this API that could be removed.